tools/config/__init__.py

```diff
@@ -610,24 +610,76 @@ def _get_mem_specs(self, memories, cmsis_part, exception_text):
                 continue
         raise ConfigException(exception_text)
 
-    @property
-    def rom(self):
-        """Get rom information as a pair of start_addr, size"""
+    def get_all_rom_regions(self):
```

Comment:
As we are now creating 2 conflicting definitions for "Regions" I recomend using the term "Memories" for these functions.

Memories, by this definition, will be any hardware defined (via cmsis-packs and targets.json overrides) places to store stuff.
Regions, by this definition, will be a partition within the available memories.
The active regions are the regions that will be used during this link step.